ivy.el

```diff
@@ -4578,10 +4578,12 @@ Otherwise, forward to `ivy-kill-line'."
     x))
 
 (defun ivy--action-insert (x)
-  (insert
-   (if (stringp x)
-       (ivy--trim-grep-line-number x)
-       x (car x))))
+  (insert (cond ((stringp x) (ivy--trim-grep-line-number x))
+                ;; swiper-isearch passes buffer pos
+                ((numberp x) (save-excursion
```

Comment:
When checking for buffer positions, a more specific predicate is usually desirable, namely `integer-or-marker-p`, since markers also represent buffer positions, but floating point numbers do not.